packages/core/src/sync-store/postgres/store.ts

```diff
@@ -823,277 +831,275 @@ export class PostgresSyncStore implements SyncStore {
-            "blocks.hash as block_hash",
-            "blocks.logsBloom as block_logsBloom",
-            "blocks.miner as block_miner",
-            "blocks.mixHash as block_mixHash",
-            "blocks.nonce as block_nonce",
-            "blocks.number as block_number",
-            "blocks.parentHash as block_parentHash",
-            "blocks.receiptsRoot as block_receiptsRoot",
-            "blocks.sha3Uncles as block_sha3Uncles",
-            "blocks.size as block_size",
-            "blocks.stateRoot as block_stateRoot",
-            "blocks.timestamp as block_timestamp",
-            "blocks.totalDifficulty as block_totalDifficulty",
-            "blocks.transactionsRoot as block_transactionsRoot",
-
-            "transactions.accessList as tx_accessList",
-            "transactions.blockHash as tx_blockHash",
-            "transactions.blockNumber as tx_blockNumber",
-            "transactions.from as tx_from",
-            "transactions.gas as tx_gas",
-            "transactions.gasPrice as tx_gasPrice",
-            "transactions.hash as tx_hash",
-            "transactions.input as tx_input",
-            "transactions.maxFeePerGas as tx_maxFeePerGas",
-            "transactions.maxPriorityFeePerGas as tx_maxPriorityFeePerGas",
-            "transactions.nonce as tx_nonce",
-            "transactions.r as tx_r",
-            "transactions.s as tx_s",
-            "transactions.to as tx_to",
-            "transactions.transactionIndex as tx_transactionIndex",
-            "transactions.type as tx_type",
-            "transactions.value as tx_value",
-            "transactions.v as tx_v",
-          ])
-          .where("logs.checkpoint", ">", encodeCheckpoint(fromCheckpoint))
-          .where("logs.checkpoint", "<=", encodeCheckpoint(toCheckpoint))
-          .orderBy("logs.checkpoint", "asc")
-          .limit(limit + 1)
-          .execute(),
-
-        this.db
-          .selectFrom("logs")
-          .leftJoin("blocks", "blocks.hash", "logs.blockHash")
-          .select(["logs.checkpoint"])
-          .where((eb) => {
-            const logFilterCmprs =
-              logFilters?.map((logFilter) => {
-                const exprs = this.buildLogFilterCmprs({ eb, logFilter });
-
-                exprs.push(eb("logs.topic0", "=", logFilter.eventSelector));
-
-                return eb.and(exprs);
-              }) ?? [];
-
-            const factoryCmprs =
-              factories?.map((factory) => {
-                const exprs = this.buildFactoryCmprs({ eb, factory });
-                exprs.push(eb("logs.topic0", "=", factory.eventSelector));
+  }) {
+    let cursor = encodeCheckpoint(fromCheckpoint);
 
-                return eb.and(exprs);
-              }) ?? [];
+    while (true) {
+      const events = await this.db.wrap(
+        { method: "getLogEvents" },
+        async () => {
+          // Get full log objects, including the eventSelector clause.
+          const requestedLogs = await this.db
```

Comment:
@jaylmiller WDYT about this query? I'm a bit worried about an explosion in row count due to the inner join as the # of sources increases.

Reply:
i think if we have their filters (i.e. whatever the equivalent to `logFilterCmprs` and `factoryCmprs`) be based on the primary row the row count should stay okay. if the row count started getting too big for a given set of filters, we could reduce the checkpoint interval which would reduce the number of rows being joined? So i think this query seems good. I'm not sure of a better way to handle fetching this data than what this query is doing, but good to keep it mind, specifically re the checkpoint interval size